src/routes/profile.js

```diff
@@ -155,30 +193,139 @@ router.get(
   },
 );
 
+const storage = multer.memoryStorage();
+const upload = multer({ storage: storage }).fields([
+  { name: 'avatarRes', maxCount: 1 },
+  { name: 'avatarSpy', maxCount: 1 },
+]);
+
 // Update the customavatar
 router.post(
   '/:profileUsername/changeavatar',
   checkProfileOwnership,
-  (req, res) => {
-    console.log('Recieved change avatar');
-    console.log(`For user ${req.params.profileUsername}`);
-    console.log(`Res link: ${req.body.reslink}`);
-    console.log(`Spy link: ${req.body.spylink}`);
-    console.log(`Message to mod: ${req.body.msgToMod}`);
+  upload,
+  async (req, res) => {
+    const MAX_AVATAR_REQUESTS = 2;
+    const MIN_GAMES_REQUIRED = 100;
+    const VALID_DIMENSIONS = [128, 1024];
 
-    // sometimes https links dont show up correctly
-    // req.body.reslink.replace("https", "http");
-    // req.body.spylink.replace("https", "http");
+    const s3 = req.s3;
+    const user = await User.findOne({ username: req.params.profileUsername });
 
-    const avatarRequestData = {
-      forUsername: req.params.profileUsername.toLowerCase(),
+    // TODO: how should i handle this?
+    if (!user) {
+      console.log('User not found');
+    }
 
-      resLink: sanitizeHtml(req.body.reslink),
-      spyLink: sanitizeHtml(req.body.spylink),
-      msgToMod: sanitizeHtml(req.body.msgToMod),
+    // Checks if custom avatar request is valid
+    if (user.totalGamesPlayed < MIN_GAMES_REQUIRED) {
+      req.flash(
```

Comment:
Instead of req.flash, let's just return the body as a string and return the appropriate http error code for the client to render it appropriately (proabably in a sweet alert).

Reply:
Have a look at this guy for the frontend on how to send a POST req from a form and render the result on screen. 

https://github.com/vck3000/ProAvalon/blob/master/assets/scripts/lobby/sockets/commands.js#L16-L71